src/tests/ftest/util/nvme_utils.py

```diff
@@ -208,7 +208,8 @@ def set_device_faulty(self, server, disk_id):
         self.dmg.storage_set_faulty(disk_id)
         result = self.dmg.storage_query_device_health(disk_id)
         # Check if device state changed to EVICTED.
-        if 'State:EVICTED' not in result.stdout_text:
+        if result['response']['host_storage_map']['storage']['smd_info']['devices']['dev_state'] \
+                != 'EVICTED':
```

Comment:
Use code similar to https://github.com/daos-stack/daos/blob/master/src/tests/ftest/nvme/health.py#L115-L133
```suggestion
        info = get_dmg_smd_info(self, dmg.storage_set_faulty, 'devices', uuid=uuid)
        for devices in info.values():
            for device in devices:
                if device['uuid'] != uuid:
                    continue
                if device['dev_state'].lower() != 'evicted':
                    self.fail("device State {} on host {} suppose to be EVICTED".format(disk_id, server))
```

Reply:
Please see the update.